### ISSUE TITLE
Mouse feels terrible

### ISSUE THREAD

**Issue by StrikerMan780** [Aleph-One-Marathon/alephone#17]
For some reason, Aleph One has this strange stiffness in the mouse. Moving the mouse slow has no difference in movement than moving it fast. You always turn at the same speed, making it extremely difficult to control. It's almost as if trying to control the game with a digital pad.

I've tried different sensitivity settings, and while I can make it turn me faster or slower, the speed is exactly constant regardless if I move the mouse a little bit, or a lot.

I hope I'm explaining this clearly, it's kind of difficult to explain. All I can say for certain, is the mouse feels really bad, worse than any mouse-controlled game I've ever played. Aleph One is a great port, don't get me wrong, but that's the one thing that prevents me from playing through the games. I can't stand it.

**Comment by LidMop** [comment]
Are you talking about the fact that the angular granularity of the player's direction is coarse? That's the only aiming issue I'm aware of. It dates back to the original Marathon engine, but I agree with you; it's janky by modern standards. You can sort of compensate by sidestepping.

Incidentally, I recently started attemping to understand the codebase with the aim of fixing this issue myself (among other issues). I don't know if fixing it is feasible, but I'll post an update when I learn more.

**Comment by StrikerMan780** [comment]
It's not the granularity, is that there's no difference in movement regardless if you move the mouse a foot, or only an inch... it seems to be based on the amount of time you move it. Feels like keyboard steering.

**Comment by LidMop** [comment]
It might be an SDL bug, then. Aleph One currently uses SDL 1.2, which is out of date and has a few mouse-related bugs that got fixed or went away in SDL 2. Hopper [mentioned](http://pfhorums.com/viewtopic.php?p=164986#p164986) that he'll be upgrading Aleph One to SDL 2 over the winter, so when that comes out it might fix your issue.

**Comment by darealshinji** [comment]
> Hopper mentioned that he'll be upgrading Aleph One to SDL 2 over the winter

 :smile: :thumbsup:

**Comment by Medicris** [comment]
This occurs to me at the low senstivities I like to use. From my understanding, the problem is the engine's limited entity angle precision: due to this, mouselook speeds only work on a threshold method. Mouse speed translates into a series of turnspeed rates ("When you move the mouse, you are affecting your angular acceleration. There is also a cap on angular velocity, which is in the physics file").

On the low end of the spectrum, this is most apparent: slow movements act as you describe, making fine aiming difficult. It's all-or-nothing, as even very small, fine movements "drift" a lot because your entity is literally incapable of turning less degrees per tic in the direction you want.

In short, pretty sure it's just the engine. SDL2's raw input is appealing, but I can't think of a way around this besides disconnecting mouselook from the ingame entity somehow. Like having the camera able to look with any degree of precision, with the player entity facing the nearest possible angle, a very tall order.

**Comment by StrikerMan780** [comment]
Even vanilla Doom had that same limited angle precision, but the mouse didn't feel terrible like that. The lack of fine granularity is to be expected, but It shouldn't ever cause the behavior I'm trying to describe.

**Comment by SolraBizna** [comment]
Have you tried both with and without "Accelerate Mouse" enabled?

**Comment by orbea** [comment]
@Hopper262 Any thoughts on how practical it would be to further improve mouse look? Even with the current master its still stiff and awkward compared to other engine ports of retro fps games. I think a smooth mouse look experience would really improve alephone for modern standards.

**Comment by orbea** [comment]
Thanks for the response and efforts!

I have SDL 2.0.9 and tried changing the settings as you suggested, this helps a lot. It does feel like it could improve a bit more still, for me the main issue is the rigidness of changing angles, its nauseating at times even.

**Comment by LidMop** [comment]
There is a **third** limitation that I only just came to understand: the game drops instead of accumulating any unused fractions of mouse deltas, e.g. a slow enough mouse movement won't register at all. #117 fixes that defect, seperately from the preferences option. I'm 95% certain that this defect is the cause of the "rigidness" you're experiencing, @orbea.

**Comment by orbea** [comment]
I've also tested PR https://github.com/Aleph-One-Marathon/alephone/pull/117 and while it somewhat feels like a step in the right direction, its also feels not quite there yet.
